osutil/kcmdline_test.go

```diff
@@ -208,3 +208,66 @@ func (s *kcmdlineTestSuite) TestKernelCommandLine(c *C) {
 	c.Assert(err, IsNil)
 	c.Check(cmd, Equals, "foo bar baz panic=-1")
 }
+
+func (s *kcmdlineTestSuite) TestKernelParseCommandLine(c *C) {
```

Comment:
Does the kernel itself has any sort of (unit) test or test-values that we could replicate here? (or maybe that happend already, if so a short comment would be great).

Reply:
Not for the `next_arg` function afaict. What I did though was to copy over the code to a small C program and checked that the outcome of the function was the same as the one I implemented in golang - which is the case minus sometimes the linux code returning NULL or "" while we always return "" in both cases, but I don't think that is relevant. Added a comment in any case.